### ISSUE TITLE
next-urql's ssr doesn't work with Next 12.1.5

### ISSUE THREAD

**Issue by arcanis** [FormidableLabs/urql#2400]
**Steps to reproduce**

Clone https://github.com/arcanis/next-bug-context-ssr, run `yarn && yarn next dev`, and open localhost. It will crash. If you run `yarn add next@12.1.5-canary.1` before restarting the server, it'll work.

I'm not sure if the actual problem comes from Next having a regression, or next-urql doing something bad, so I figured I should open an issue on both trackers - the Next.js counterpart is https://github.com/vercel/next.js/issues/36198.

**Expected behavior**

Should work.

**Actual behavior**

I have some contexts defined inside layout components (using a pattern similar to the one described [here](https://nextjs.org/docs/basic-features/layouts)), and since this upgrade those contexts are missing in the SSR build, crashing the direct page load (navigating to the page client-side works fine). It's almost as if pages were built without the _app wrapper.

**Comment by pdong** [comment]
I ran into the same issue as you and narrowed it down to a change in canary.1 and canary.2 as well.  I was able to get it working by disabling Suspense via the options in `withUrqlClient`. e.g.

```
{ ssr: true, neverSuspend: true }
```

I haven't dug into it too much but it seems like something changed and prepass isn't correctly catching the errors anymore?  LMK if `neverSuspend` fixes your issue for now though; it seems to have resolved my inability to access context when doing SSR
